Foundation/NSGeometry.swift

```diff
@@ -508,12 +508,84 @@ public func NSIntersectsRect(aRect: NSRect, _ bRect: NSRect) -> Bool {
         NSMaxX(aRect) <= NSMinX(bRect) || NSMaxX(bRect) <= NSMinX(aRect) || NSMaxY(aRect) <= NSMinY(bRect) || NSMaxY(bRect) <= NSMinY(aRect))
 }
 
-public func NSStringFromPoint(aPoint: NSPoint) -> String { NSUnimplemented() }
-public func NSStringFromSize(aSize: NSSize) -> String { NSUnimplemented() }
-public func NSStringFromRect(aRect: NSRect) -> String { NSUnimplemented() }
-public func NSPointFromString(aString: String) -> NSPoint { NSUnimplemented() }
-public func NSSizeFromString(aString: String) -> NSSize { NSUnimplemented() }
-public func NSRectFromString(aString: String) -> NSRect { NSUnimplemented() }
+public func NSStringFromPoint(aPoint: NSPoint) -> String {
+    return "{\(aPoint.x.native), \(aPoint.y.native)}"
+}
+
+public func NSStringFromSize(aSize: NSSize) -> String {
+    return "{\(aSize.width.native), \(aSize.height.native)}"
+}
+
+public func NSStringFromRect(aRect: NSRect) -> String {
+    let originString = NSStringFromPoint(aRect.origin)
+    let sizeString = NSStringFromSize(aRect.size)
+    
+    return "{\(originString), \(sizeString)}"
+}
+
+private func _scanTwoNumbersFromGeometryString(aString: String) -> (first: Double, second: Double) {
+    guard aString.hasPrefix("{") && aString.hasSuffix("}") else {
+        fatalError("Wrong format")
+    }
+    
+    var string: NSString = aString._nsObject
+    string = string.substringWithRange(NSMakeRange(1, aString.length - 2))._nsObject
+    string = string.stringByReplacingOccurrencesOfString(" ", withString: "")._nsObject;
+    
+    let numbers = string.componentsSeparatedByString(",")
```

Comment:
It is because I have run across silliness like that before in code. Specifically the one I saw in the wild was someone was creating points as the format 4x5; which is definitely in the territory of undocumented/undefined behavior.